### PR TITLE
More materialize documentation

### PR DESCRIPTION
**Pull request by eric-anderson**
Copy over documentation from docset to transforms/materialize.md Copy some examples from the upcoming blog post.

### CODE REVIEW

File: docs/source/sycamore/transforms/materialize.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+The `materialize` transform writes out documents up to that point, marks the materialized path
```
Comment (HenryL27) [inline docs/source/sycamore/transforms/materialize.md:1]: title line?

File: docs/source/sycamore/transforms/materialize.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+as successful if execution is successful, and allows for reading from the materialized data as
+a source.
+
```
Comment (jonfritz) [inline docs/source/sycamore/transforms/materialize.md:4]: This is helpful if you're using a notebook and using show() and take() functions to inspect your data in the middle of a Sycamore script. Using `materialize` prevents Sycamore from recomputing after these functions.

**Review by jonfritz — COMMENTED** [review]
2 comments

File: docs/source/sycamore/transforms/materialize.md (hunk 1 above)
Comment (eric-anderson) [reply docs/source/sycamore/transforms/materialize.md:3]: done.

File: docs/source/sycamore/transforms/materialize.md (hunk 2 above)
Comment (eric-anderson) [reply docs/source/sycamore/transforms/materialize.md:7]: done.

**Review by HenryL27 — APPROVED** [review]

**Pull request merged**